### ISSUE TITLE
Moodle 4.3 compatibility issue

### ISSUE THREAD

**Issue by Buda9** [kamenskiyyyy/moodle-moyklass-plugin#3]
On the page /local/moyclass/index.php when using Moodle 4.3 I see this error.

```
Manage ForClasses

Exception - local_moyclass\api_service::get_lessons(): Return value must be of type array, null returned

Debug info:
Error code: generalexceptionmessageDismiss this notification
Stack trace:
line 123 of \local\moyclass\classes\api_service.php: TypeError thrown
line 241 of \local\moyclass\classes\manager_db.php: call to local_moyclass\api_service->get_lessons()
line 43 of \local\moyclass\classes\pages\dashboard.php: call to local_moyclass\manager_db->set_lessons()
line 41 of \local\moyclass\index.php: call to local_moyclass\pages\dashboard->render()
```

Pointing to https://github.com/kamenskiyyyy/moodle-moyklass-plugin/blob/master/classes/api_service.php#L117

I'm not sure how to solve it by myself. Could you please take a quick look @kamenskiyyyy?

PS. Yes, there are some non-Russian schools using MoyKlass :)

**Comment by kamenskiyyyy** [comment]
@Buda9 Hi! 
Unfortunately, I am no longer supporting this plugin and have not tested its functionality at the moment. Perhaps the MoyKlass api has changed…

**Comment by Buda9** [comment]
@kamenskiyyyy, I greatly appreciate your response. The issue turned out to be a PHP incompatibility, as I was using PHP 8.2. It's all resolved now. If you think it's helpful, I can submit a pull request (PR)?

Additionally, I have few questions where your insight would be invaluable since you're the most knowledgeable about it.

I've noticed the plugin includes various pages (like lessons, subscriptions, teacher, etc.) and widgets. However, I've only been able to access three pages so far:
- /local/moyclass/index.php
- /local/moyclass/lessons.php and 
- /local/moyclass/subscriptions.php

When I wanted to add widgets (are these "block" widgets?), I don't see any available.

Also, when user register using moyklass widget form, is he supposed to be added to Moodle too? Is this one of plugin features or I misunderstood it? (Edit: I cannot yet test this because Cron is throwing errors and not running because of php incompatibility issues)

In summary, as you can see, I'm trying to understand how the plugin works so I can enhance it a little bit (a lot of work is already done) and share it so anyone can utilize power of moyklass with Moodle
